### PR TITLE
Redo iOS UI to be programmatic instead of storyboard based & fix layering in UI app to separate UI with business logic

### PR DESCRIPTION
**Pull request by shana-apple**
#### Problem
The current demo app is storyboard based which makes it not only difficult to modify and add to but makes it impossible for multiple people to add to the UI at the same time. 
The UI code was also mixed with business logic instead of having clean layering.

 #### Summary of Changes
Remove storyboard and redo all UI programmatically
Fix layering when needed and move business logic out of UI View Controllers (those classes should ideally only contain UI manipulation)
Add utils to avoid duplicated code

fixes #<issue number, or numbers. if no issue, please create one>

### CODE REVIEW

**Review by woody-apple — APPROVED** [review]

File: src/darwin/CHIPTool/CHIPTool.xcodeproj/project.pbxproj [hunk 1]
```diff
@@ -525,6 +550,7 @@
 			buildSettings = {
 				BUNDLE_LOADER = "$(TEST_HOST)";
 				CODE_SIGN_STYLE = Automatic;
+				DEVELOPMENT_TEAM = "";
```
Comment (sagar-apple) [inline src/darwin/CHIPTool/CHIPTool.xcodeproj/project.pbxproj:553]: Let's leave this setting out?

File: src/darwin/CHIPTool/CHIPTool/View Controllers/Echo client/EchoViewController.m [hunk 2]
```diff
@@ -0,0 +1,225 @@
...
+    } else {
+        stringMessage = [[NSString alloc] initWithData:message encoding:NSUTF8StringEncoding];
+    }
+    NSString *resultMessage = [@"Echo Response: " stringByAppendingFormat:@"%@", stringMessage];
+    dispatch_after(dispatch_time(DISPATCH_TIME_NOW, 5.0), dispatch_get_main_queue(), ^{
+        [self updateResult:resultMessage];
+    });
+}
+
+// MARK: CHIPConnectivityManagerDelegate
```
Comment (sagar-apple) [inline src/darwin/CHIPTool/CHIPTool/View Controllers/Echo client/EchoViewController.m:202]: did you mean `#pragma mark` or does this end up doing the same thing?

File: src/darwin/Framework/CHIP.xcodeproj/project.pbxproj [hunk 3]
```diff
@@ -435,6 +435,7 @@
 			isa = XCBuildConfiguration;
 			buildSettings = {
 				CHIP_ROOT = "$(SRCROOT)/../../..";
+				CODE_SIGN_IDENTITY = "Apple Development";
```
Comment (sagar-apple) [inline src/darwin/Framework/CHIP.xcodeproj/project.pbxproj:438]: Let's not add this either.

File: src/darwin/CHIPTool/CHIPTool/View Controllers/Echo client/EchoViewController.m (hunk 2 above)
Comment (shana-apple) [reply src/darwin/CHIPTool/CHIPTool/View Controllers/Echo client/EchoViewController.m:202]: // MARK: is swift and objc compatible vs pragma mark is for objective c 

I always advised to use MARK on my old team to keep consistent between the objc/swift files so I keep doing that

Comment (sagar-apple) [reply src/darwin/CHIPTool/CHIPTool/View Controllers/Echo client/EchoViewController.m:202]: TIL, thanks!

**Review by BroderickCarlin — APPROVED** [review]

File: src/darwin/CHIPTool/CHIPTool/UI Helpers/CHIPUIViewUtils.m [hunk 4]
```diff
@@ -0,0 +1,142 @@
...
+    
+    return titleLabel;
+}
+
++ (UIStackView *)stackViewWithLabel:(UILabel *)label result:(UILabel *)result
+{
+    UIStackView *stackView = [UIStackView new];
+    stackView.axis = UILayoutConstraintAxisVertical;
+    stackView.distribution = UIStackViewDistributionEqualSpacing
+;
```
Comment (mspang) [inline src/darwin/CHIPTool/CHIPTool/UI Helpers/CHIPUIViewUtils.m:43]: move semicolon to previous line?

**Review by mspang — APPROVED** [review]

**Review by rwalker-apple — APPROVED** [review]

**Review by sagar-apple — APPROVED** [review]

**Pull request merged**